### PR TITLE
WFCORE-310 - change ignore-unused-configuration default true

### PR DESCRIPTION
**Pull request by luck3y**
WFCORE-310 - WFCORE-310 default ignore-unused-configuration to true unless --backup is used, or a value is supplied in host.xml

### CODE REVIEW

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/suites/IgnoredResourcesTestCase.java [hunk 1]
```diff
@@ -75,6 +83,31 @@ public static void setupDomain() throws Exception {
...
+        domainSlaveLifecycleUtil.connect();
+        domainSlaveLifecycleUtil.awaitHostController(System.currentTimeMillis());
+
+        domainMasterLifecycleUtil = testSupport.getDomainMasterLifecycleUtil();
+        domainSlaveLifecycleUtil = testSupport.getDomainSlaveLifecycleUtil();
+
+        ModelNode slaveIgnore = Util.createEmptyOperation(READ_RESOURCE_OPERATION, PathAddress.pathAddress(HOST, "slave"));
+        ModelNode result = DomainTestUtils.executeForResult(slaveIgnore, domainSlaveLifecycleUtil.getDomainClient());
+        // make sure that ignore-unused-configuration is false
+        Assert.assertFalse(result.get(DOMAIN_CONTROLLER).get(REMOTE).get(IGNORE_UNUSED_CONFIG).asBoolean());
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/suites/IgnoredResourcesTestCase.java]: This needs to be reversed when the test is done, or the rest of the suite will execute with an unexpected config.

After CR2 we should consider separating this test out from DomainTestSuite and just giving it its own config files. Part of the reason for that is a fair bit of the stuff in the config files DomainTestSuite uses may only be there for this test. I'm not sure about that though; it's just a thought.

**Comment by bstansberry** [comment]
Besides the comment about reversing the ignore-unused-configuration=false change when the test is done, the end result looks good. Please squash the commits though.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by kabir** [comment]
Superseded by https://github.com/wildfly/wildfly-core/pull/1091

**Pull request closed without merging**